DPL: handle corner case for consumeWhenAll

ktf opened the pull request:
DPL: handle corner case for consumeWhenAll

This should avoid computation if only sporadic inputs are expected
and they are all missing.

ktf commented:
@martenole @shahor02 as discussed on mattermost. This will skip the case N/N sporadic inputs are missing (and no timeframe data is requested).
   
Notice this will still schedule the case in which some of N sporadic inputs are missing, so that's supposed to be protected.

shahor02 commented on Framework/Core/src/CompletionPolicyHelpers.cxx:
```diff
@@ -145,11 +154,15 @@ CompletionPolicy CompletionPolicyHelpers::consumeWhenAll(const char* name, Compl
     }
     O2_SIGNPOST_END(completion, sid, "consumeWhenAll", "Completion policy returned %{public}s for timeslice %lu <= oldestPossibleTimeslice %lu", needsProcessing ? "Consume" : "Discard", currentTimeslice, oldestPossibleTimeslice);
+    // No need to process if we have only sporadic inputs and they are all missing.
+    if (needsProcessing && (missingSporadicCount == sporadicCount) && (timeframeCount == 0)) {
+      return CompletionPolicy::CompletionOp::Discard;
+    }
     return needsProcessing ? CompletionPolicy::CompletionOp::Consume : CompletionPolicy::CompletionOp::Discard;
```
```suggestion
    if (needsProcessing && (sporadicCount > 0) && (missingSporadicCount == sporadicCount) && (timeframeCount == 0)) {
      return CompletionPolicy::CompletionOp::Discard;
    }
    return (needsProcessing ||  sporadicCount == 0) ? CompletionPolicy::CompletionOp::Consume : CompletionPolicy::CompletionOp::Discard;
```

ktf committed:
DPL: handle corner case for consumeWhenAll

This should avoid computation if only sporadic inputs are expected
and they are all missing.
